LuckyDragon/app/src/main/java/com/example/luckydragon/OrganizerProfileFragment.java

```diff
@@ -4,67 +4,53 @@
 
-import static android.content.ContentValues.TAG;
-
 import android.app.Activity;
 import android.os.Bundle;
 import android.util.Log;
 import android.view.View;
 import android.widget.Button;
 import android.widget.ImageButton;
 import android.widget.ListView;
+import android.widget.TextView;
 
 import androidx.annotation.NonNull;
 import androidx.fragment.app.DialogFragment;
 import androidx.fragment.app.Fragment;
 
 import com.google.android.material.textview.MaterialTextView;
-import com.google.firebase.firestore.FirebaseFirestore;
-import com.google.firebase.firestore.QueryDocumentSnapshot;
-
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
 
 public class OrganizerProfileFragment extends Fragment {
-    private FirebaseFirestore db = FirebaseFirestore.getInstance();
-    private String facilityName;
-    Set<Event> eventSet = new HashSet<>();
-    ArrayList<Event> eventList;
-    EventArrayAdapter eventListAdapter;
+    private OrganizerProfileView organizerProfileView;
+    private User user;
+    private EventArrayAdapter eventListAdapter;
 
     public OrganizerProfileFragment() {
         super(R.layout.fragment_organizer_profile);
     }
 
     @Override
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
-        // Get facility from parent activity
-        ProfileActivity parent = (ProfileActivity)requireActivity();
-        User user = parent.getUser();
+        // Get user
+        user = ((GlobalApp) requireActivity().getApplication()).getUser();
+        assert user.getOrganizer() != null; // user must be an organizer by this point
+        user.getOrganizer().fetchEvents();
 
-        assert user.getOrganizer() != null;  // if this runs, we messed up in coding.
-        facilityName = user.getOrganizer().getFacility();
+        // Set up organizer events listview
+        ListView eventsListView = view.findViewById(R.id.organizerProfileEventsListview);
+        eventListAdapter = new EventArrayAdapter(user.getOrganizer().getEvents(), requireActivity().getApplicationContext());
+        eventsListView.setAdapter(eventListAdapter);
 
-        // Set facility name
-        MaterialTextView facilityTextView = view.findViewById(R.id.facilityTextView);
-        ImageButton facilityEditButton = view.findViewById(R.id.facilityEditButton);
-        if (facilityName == null) {
-            // Set facility text
-            String noFacilityMessage = "You have not created a facility yet.";
-            facilityTextView.setText(noFacilityMessage);
-            // Change button to "add" icon
-            facilityEditButton.setImageResource(R.drawable.baseline_add_24);
-        } else {
-            // Set facility text
-            facilityTextView.setText(facilityName);
-        }
+        // Create view
+        organizerProfileView = new OrganizerProfileView(user, this);
+
+        // Update view
+        setFacilityName();
+        updateEventsList();
 
```

Comment:
Should be inside the organizerProfileView. Note that update is called as soon as you create the view